specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,260 @@
+
+Bring back non-visible card element            |  Bring back single input
+:-------------------------:|:-------------------------:
+![img](assets/InputValidation/BrokenToggleWizard.png)  |  ![img](assets/InputValidation/BrokenToggleWizard2.png)
+
+Additionally, per the below discussion, if we disable the submit button when validation fails, we won't event have an appropriate time at which to change the visibility.
+
+> Note on this card: This "Wizard" scenario is implemented here via toggle visibility. We should consider whether we want to support a first class wizard experience where a new element type could list a series of subcards to display in order. Whether or not we ever introduce such a concept, the point remains that toggle logic may be complex, and we shouldn't be arbitrarily toggling things visible outside the scope of what the card author has defined in their toggle buttons.
+
+#### Current Card/Container
+
+Considering the food order card above, one option in show card scenarios may be to validate the current show card, or the show card and any parent cards. This works well for the food order card, but doesn't help us with card scenarios that include toggles. It may also not be the correct behavior for a mid card action set containing a show card. While it would be nice if we could provide an algorithm that deduces which inputs should be validated, it would be tricky to create an algorithm that gracefully handles all cases. Additionally, the more complicated the algorithm, the harder it is for card authors to understand the expected behavior.
+
+#### Card Author Control
+
+As the discussion above shows, depending on the card scenario, it may be difficult for us to determine which fields the card author wants to validate. Any of the above approaches may be correct for *some* cards, but none are appropriate for all cards. In order to address this, the best approach is to allow the card author to specify which inputs are associated with a given action. We will implement this via a new `inputs` property on Action types. The inputs listed will be the ones that are validated, and also the ones that are returned on Submit.
+
+The first option available for inputs is "All." This will validate and submit all inputs on the card. This is the current behavior, and would be the default for submit actions.
+```json
+{
+	"type": "Action.Submit",
+	"title": "Submit",
+	"inputs": "All",
+}
+```
+
+The second option would be "None". This would be useful for, for example, a "Cancel" button where validation is not required, and would be the default for non-Submit (or Invoke action being discussed for Universal Actions) action types (see further discussion of validation for different action types below).
+```json
+{
+	"type": "Action.Submit",
+	"title": "Cancel",
+	"inputs": "None",
```

Comment:
>"inputs": "None", [](start = 1, length = 17)

Discussion: What is the scenario for "None"? Should we have an explicit Action.Cancel instead?
- Fire company has an Action.Cancel in their stuff, but what are the semantics?
- Non-submit action types default to None.